sdk/typescript/jest.config.js

```diff
@@ -0,0 +1,6 @@
+/** @type {import('@jest/types').Config.InitialOptions} */
+const config = {
+  testEnvironment: 'node',
```

Comment:
The default jest config uses a `jsdom`-based test environment which is not very accurate and does not include the `TextEncoder` global. The tests shouldn't need anything DOM related so they can use the bare node environment.